### DISCUSSION TITLE
gettext libintl doesn't work correctly

### DISCUSSION THREAD

**Discussion by nrbrook** [Homebrew/discussions#109] [Everyday usage]
In a C App, after calling `setlocale` and `bindtextdomain`, `dgettext` doesn't return a string from the specified locale message catalogue. Instead, libintl uses the locale defined in the `LANG` environment variable. The only way to change the locale is by changing `LANG`: `putenv((char *)"LANG=be_BY");`. See the [example project](https://github.com/Homebrew/discussions/files/5470951/untitled1.zip) and the [stack overflow question](https://stackoverflow.com/questions/64572654/dgettext-failing-to-translate-string/64613742). I'm not sure where this problem lies – `libintl`? The homebrew formulae?

**Comment by SMillerDev** [comment]
Did you try contacting the gettext project to verify this is actually a packaging error and not just an upstream bug? Because we don't provide code support so you'll have to come up with a test that would make this fail in the homebrew test block.

**Reply by nrbrook** [reply]
I’ll try that. Thanks
